packages/flutter_tools/lib/src/build_system/targets/assets.dart

```diff
@@ -343,6 +345,11 @@ class CopyAssets extends Target {
 
   @override
   Future<void> build(Environment environment) async {
+    final String? buildModeEnvironment = environment.defines[kBuildMode];
+    if (buildModeEnvironment == null) {
+      throw MissingDefineException(kBuildMode, name);
```

Comment:
i wonder if we'll ever hit this. merging this and then seeing if we need to roll back sounds good to me, though.